src/Passage.sol

```diff
@@ -106,54 +95,6 @@ contract Passage {
         enterToken(defaultRollupChainId, rollupRecipient, token, amount);
     }
 
-    /// @notice Allows a special transaction to be sent to the rollup with sender == L1 msg.sender.
-    /// @dev Transaction is processed after normal rollup block execution.
-    /// @dev See `enterTransact` for docs.
-    function transact(
-        uint256 rollupChainId,
-        address to,
-        bytes calldata data,
-        uint256 value,
-        uint256 gas,
-        uint256 maxFeePerGas
-    ) public payable {
-        enterTransact(rollupChainId, msg.sender, to, data, value, gas, maxFeePerGas);
-    }
-
-    /// @dev See `transact` for docs.
-    function transact(address to, bytes calldata data, uint256 value, uint256 gas, uint256 maxFeePerGas)
-        external
-        payable
-    {
-        enterTransact(defaultRollupChainId, msg.sender, to, data, value, gas, maxFeePerGas);
-    }
-
-    /// @notice Send Ether on the rollup, send a special transaction to be sent to the rollup with sender == L1 msg.sender.
-    /// @dev Enter and Transact are processed after normal rollup block execution.
-    /// @dev See `enter` for Enter docs.
-    /// @param rollupChainId - The rollup chain to send the transaction to.
-    /// @param etherRecipient - The recipient of the ether.
-    /// @param to - The address to call on the rollup.
-    /// @param data - The data to send to the rollup.
-    /// @param value - The amount of Ether to send on the rollup.
-    /// @param gas - The gas limit for the transaction.
-    /// @param maxFeePerGas - The maximum fee per gas for the transaction (per EIP-1559).
-    /// @custom:emits Transact indicating the transaction to mine on the rollup.
-    function enterTransact(
```

Comment:
note: moving `transact` to Zenith && splitting Passage from Zenith means that one can no longer atomically `enter` + `transact`

Reply:
this seems bad

Reply:
the only ways around this would be 
- keep Passage & Zenith deployed at the same address (as we discussed, makes potential migration path worse) 
- call `Passage.enter` as an external contract (i find this to be undesirable but it could be done)

Reply:
> call Passage.enter as an external contract (i find this to be undesirable but it could be done)

seems fine